src/wolfscp.c

```diff
@@ -1035,15 +1042,15 @@ static int GetScpTimestamp(WOLFSSH* ssh, byte* buf, word32 bufSz,
     return ret;
 }
 
-
+#ifndef NO_FILESYSTEM
 /* checks for if directory is being renamed in command
  *
  * returns WS_SUCCESS on success
  */
 static int ScpCheckForRename(WOLFSSH* ssh, int cmdSz)
 {
     /* case of file, not directory */
-    char buf[cmdSz + 4];
+    char buf[DEFAULT_SCP_MSG_SZ];
```

Comment:
Please add a sanity check on cmdSz before using it in WSTRNCPY with the limited buf size now.

Reply:
Added check for the limit on the buf size (against cmdSz + 4)